roles/pulp/tasks/main.yaml

```diff
@@ -34,20 +115,43 @@
   ansible.builtin.systemd:
     daemon_reload: true
 
-- name: Start the Pulp Service
+- name: Start the Pulp API services
   ansible.builtin.systemd:
-    name: pulp
+    name: pulp-api
     enabled: true
-    state: restarted
+    state: started
+
+- name: Migrate the Pylp database
+  containers.podman.podman_container_exec:
+    name: "{{ pulp_api_container_name }}"
+    command: pulpcore-manager migrate --noinput
```

Comment:
I wonder if this should be an own command (like now), an own service (like https://github.com/pulp/pulp-oci-images/blob/latest/images/compose/compose.yml does it) or whether we should use `pulpcore-manager migrate --noinput && pulp-api` as the command for the "normal" API container startup (like we do for Foreman)?

Reply:
I guess while combining the migration and API startup commands seems convenient, but it will create challenges in error handling and clearly not flexible. 
And I think having a dedicated migration service would be a better approach for us. It allows for more flexible scaling of the migration process independently from the API and other services, making everything easier to manage in the long run.

Reply:
I see some benefits in having tighter control, but you must migrate the DB before starting any of the Pulp services.

Reply:
Fine for now but I think we should understand and decide how we handle this in a universal way given we have multiple applications having to do the same workflow: wait on database, apply outstanding migrations. Candlepin does this via the application startup procedure and so this will happen within the container itself.

Pulp's images have a wait for database and wait for migrations script that they currenly rely upon before startup: https://github.com/pulp/pulp-oci-images/blob/d74b7bbb9550152c0de1e45f16eec73f674e8f15/images/assets/pulp-api#L3-L4

The Foreman container being used takes a similar approach: https://github.com/theforeman/foreman-quadlet/blob/98f6a4110998d2f36257b9b18415b76571a2f6c4/container-images/foreman/Containerfile#L18